### PR TITLE
Module Runtime refactor + EST fixes

### PR DESCRIPTION
**Pull request by massand**

### CODE REVIEW

File: identity/aziot-identityd/src/identity.rs [hunk 1]
```diff
@@ -82,6 +82,54 @@ impl IdentityManager {
...
+					&*curr_module.module_id,
+						Some(aziot_identity_common::hub::AuthMechanism {
+							symmetric_key: Some(aziot_identity_common::hub::SymmetricKey { 
+								primary_key: Some(http_common::ByteString(primary_key)), 
+								secondary_key: Some(http_common::ByteString(secondary_key)),
+							}),
+							x509_thumbprint: None, 
+							type_: Some(aziot_identity_common::hub::AuthType::Sas),
+						}),
+		None).await
```
Comment (daprilik) [inline identity/aziot-identityd/src/identity.rs:117]: [-- 👀 --](https://github.com/rust-lang/rustfmt)

File: identity/aziot-identity-common/src/lib.rs [hunk 2]
```diff
@@ -44,8 +44,18 @@ pub struct AuthenticationInfo {
...
-	X509,
+    Sas,
+    X509,
+}
+
+impl std::fmt::Display for AuthenticationType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match *self {
+            AuthenticationType::Sas => "Sas",
+            AuthenticationType::X509 => "X509",
```
Comment (gordonwang0) [inline identity/aziot-identity-common/src/lib.rs:55]: Nit: In the HTTP payloads, these will be "sas" and "x509" because of the serde rename_all to lowercase. So perhaps the casing here should be consistent with that.

File: identity/aziot-identity-common/src/lib.rs [hunk 3]
```diff
@@ -89,8 +99,8 @@ pub enum Credentials {
 impl From<Credentials> for AuthenticationInfo {
 	fn from(c: Credentials) -> Self {
 		match c {
-			Credentials::SharedPrivateKey(k) => AuthenticationInfo {
-				auth_type: AuthenticationType::SaS,
+		    Credentials::SharedPrivateKey(k) => AuthenticationInfo {
```
Comment (gordonwang0) [inline identity/aziot-identity-common/src/lib.rs:102]: Nit: spacing

File: identity/aziot-identityd/src/identity.rs (hunk 1 above)
Comment (massand) [reply identity/aziot-identityd/src/identity.rs:117]: I read that and other spacing nits as: \*cough\* rustfmt \*cough\*

File: cert/aziot-certd/src/lib.rs [hunk 4]
```diff
@@ -214,6 +215,7 @@ fn create_cert<'a>(
 			}
 
 			let mut x509 = openssl::x509::X509::builder().map_err(|err| Error::Internal(InternalError::CreateCert(Box::new(err))))?;
+			x509.set_version(version).map_err(|err| Error::Internal(InternalError::CreateCert(Box::new(err))))?;
```
Comment (arsing) [inline cert/aziot-certd/src/lib.rs:218]: Fix this in `pkcs11-test` too please.

https://github.com/Azure/iot-identity-service/blob/9d5bfea9d6586452993ce9925c260ce3d1249459/pkcs11/pkcs11-test/src/main.rs#L204

```diff
	let mut builder = openssl::x509::X509::builder()?;
+
+	builder.set_version(2)?;
```

**Review by arsing — CHANGES_REQUESTED** [review]

File: identity/aziot-identity-common/src/lib.rs (hunk 2 above)
Comment (massand) [reply identity/aziot-identity-common/src/lib.rs:55]: Removed it since it wasn't used.

File: identity/aziot-identity-common/src/lib.rs (hunk 3 above)
Comment (massand) [reply identity/aziot-identity-common/src/lib.rs:102]: Fixed

File: cert/aziot-certd/src/lib.rs (hunk 4 above)
Comment (massand) [reply cert/aziot-certd/src/lib.rs:218]: Added

**Review by arsing — APPROVED** [review]
Signing off for the certd and pkcs11-test changes.

**Review by gordonwang0 — APPROVED** [review]

File: cert/aziot-cert-client-async/src/lib.rs [hunk 5]
```diff
@@ -118,19 +118,19 @@ where
...
 {
 	let req =
-		hyper::Request::builder()
-		.method(method)
-		.uri(uri);
-	let req =
-		if let Some(body) = body {
-			let body = serde_json::to_vec(body).expect("serializing request body to JSON cannot fail").into();
-			req
+		body.map_or(
```
Comment (arsing) [inline cert/aziot-cert-client-async/src/lib.rs:121]: Make a separate PR for this please.
